packages/nouns-contracts/contracts/governance/ExcessETHBurner.sol

```diff
@@ -0,0 +1,247 @@
+     * Nouns IDs are allowed to burn excess ETH: 1100, 1200, 1300, etc.
+     *
+     * See `excessETH()` for more information on how excess ETH is defined.
+     * @dev Reverts when auction house has not yet minted the next Noun ID at which the burn is allowed.
+     */
+    function burnExcessETH() public returns (uint256 amount) {
+        if (auction.auction().nounId < nextBurnNounID) revert NotTimeToBurnYet();
+
+        amount = excessETH();
+        if (amount == 0) revert NoExcessToBurn();
+
+        IExecutorV3(owner()).burnExcessETH(amount);
+
+        nextBurnNounID += minNewNounsBetweenBurns;
+    }
+
+    /**
+     * @notice Get the amount of excess ETH in the Nouns treasury.
+     * Excess ETH is defined as the difference between the current treasury value denominated in ETH,
+     * and the expected treasury value denominated in ETH, which is defined as mean auction price * adjusted total supply.
+     * It returns zero during the waiting period, and when expected treasury value is greater than current value.
+     * If treasury balance in native ETH is less than the excess ETH calculation, it returns the treasury ETH balance.
+     * @dev This version does not support burning excess other than in native ETH, e.g. stETH, rETH, etc.
+     * @dev Reverts if there is not enough auction history to calculate the mean auction price.
+     */
+    function excessETH() public view returns (uint256) {
+        uint256 expectedTreasuryValue = expectedTreasuryValueInETH();
+        uint256 treasuryValue = treasuryValueInETH();
+
+        if (expectedTreasuryValue >= treasuryValue) return 0;
+
+        return min(treasuryValue - expectedTreasuryValue, owner().balance);
+    }
+
+    /**
+     * @notice Get the expected treasury value denomiated in ETH.
+     * Expected value is defined as mean auction price * adjusted total supply.
+     */
+    function expectedTreasuryValueInETH() public view returns (uint256) {
+        return meanAuctionPrice() * dao.adjustedTotalSupply();
+    }
+
+    /**
+     * @notice Get the current treasury value denomiated in ETH.
+     * In addition to native ETH, it also includes stETH, wETH, and rETH balances.
+     * @dev for rETH, it uses the getEthValue() function to convert rETH to ETH.
+     */
+    function treasuryValueInETH() public view returns (uint256) {
+        address owner_ = owner();
+        return owner_.balance + stETH.balanceOf(owner_) + wETH.balanceOf(owner_) + rETHBalanceInETH(owner_);
+    }
+
+    /**
+     * @notice Get the mean auction price of the last `numberOfPastAuctionsForMeanPrice` auctions.
+     * @dev Reverts if there is not enough auction history to calculate the mean auction price.
+     */
+    function meanAuctionPrice() public view returns (uint256) {
+        uint16 numberOfPastAuctionsForMeanPrice_ = numberOfPastAuctionsForMeanPrice;
+        INounsAuctionHouseV2.Settlement[] memory settlements = auction.prices(numberOfPastAuctionsForMeanPrice_);
```

Comment:
@eladmallel maybe we should add a function that returns only the prices instead the entire settlement structs.
it will take up less gas, and may be useful for other downstream applications

Reply:
fixed: 23b84bafa3f9b0105100cb7ac9b7aa0a42d4adbe